hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstWalker.java

```diff
@@ -923,6 +922,18 @@ public void visitConversion(Conversion conversion) {
 		);
 	}
 
+	@Override
+	public void visitMemberOfPredicate(MemberOfPredicate memberOfPredicate) {
+		memberOfPredicate.getLeftHandExpression().accept( this );
+		if ( memberOfPredicate.isNegated() ) {
+			appendSql( " not " );
+		}
+		appendSql( " in (" );
+
+		visitQuerySpec( memberOfPredicate.getQuerySpec() );
+		appendSql( ")" );
```

Comment:
Is there possibility that `visaitQuerySpec` adds nothing to the SQL and an erroneous SQL like `in ()` is issued (which only works for h2 but fails for any other dbs)? That is my first contribution to Hibernate and the solution is simply to change it to `in ( null )`.

Reply:
This particular one, no. For a member-of predicate it would be illegal for the sub-query to be empty.

However, you have a great point about other cases for in-predicates in the source query. I'm not sure in(null) works on all databases - seems we have had this discussion before a few years ago.